libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -756,69 +768,66 @@ AwsIotShadowError_t AwsIotShadow_DeleteAsync( IotMqttConnection_t mqttConnection
                                               const AwsIotShadowCallbackInfo_t * pCallbackInfo,
                                               AwsIotShadowOperation_t * const pDeleteOperation )
 {
-    IOT_FUNCTION_ENTRY( AwsIotShadowError_t, AWS_IOT_SHADOW_STATUS_PENDING );
+    AwsIotShadowError_t status = AWS_IOT_SHADOW_SUCCESS;
     _shadowOperation_t * pOperation = NULL;
 
     /* Check that AwsIotShadow_Init was called. */
     if( _checkInit() == false )
     {
-        IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_NOT_INITIALIZED );
+        status = AWS_IOT_SHADOW_NOT_INITIALIZED;
     }
 
-    /* Validate the Thing Name and flags for Shadow DELETE. */
-    status = _validateThingNameFlags( SHADOW_DELETE,
-                                      pThingName,
-                                      thingNameLength,
-                                      flags,
-                                      pCallbackInfo,
-                                      pDeleteOperation );
-
-    if( status != AWS_IOT_SHADOW_SUCCESS )
+    if( status == AWS_IOT_SHADOW_SUCCESS )
```

Comment:
This could be also be an else. Same for a few other instances of the same pattern below